### PR TITLE
control-service: OAuth Creds in Secrets Service

### PR DESCRIPTION
**Pull request by dakodakov**
Add support for setting Team Ouath Credentials in
the VaultJobSecretsService

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 1]
```diff
@@ -72,19 +76,23 @@ public void updateJobSecrets(String jobName, Map<String, Object> secrets) {
...
   }
 
   @Override
-  public Map<String, Object> readJobSecrets(String jobName) throws JsonProcessingException {
-    checkJobName(jobName);
+  public Map<String, Object> readJobSecrets(String teamName, String jobName)
+      throws JsonProcessingException {
+    checkInputs(teamName, jobName);
+
+    String secretKey = teamName + "/" + jobName;
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:92]: Extract the secret ket building in a separate method.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 2]
```diff
@@ -96,9 +104,59 @@ public Map<String, Object> readJobSecrets(String jobName) throws JsonProcessingE
...
+        new VaultTeamCredentials(teamName, clientId, clientSecret);
+
+    vaultOperations.opsForVersionedKeyValue(SECRET).put(secretKey, teamCredentials);
+  }
+
+  @Override
+  public VaultTeamCredentials readTeamOauthCredentials(String teamName) {
+    checkInputs(teamName);
+
+    String secretKey = teamName + "/" + TEAM_OAUTH_CREDENTIALS;
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:123]: same here

**Review by mivanov1988 — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 1 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 2 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed

**Pull request merged**